File: Cpp/1422_Maximum_Score_After_Splitting_a_String.cpp
class Solution {
public:
    int maxScore(string s) {
        int result = 0;
        int one = 0;
        int zero = 0;

        for(auto& ch : s) {
            if(ch == '1') ++one;
        }

        for(int i = 0; i < s.size() - 1; ++i) {
            if(s[i] == '0') ++zero;
            if(s[i] == '1') --one;

            result = max(result, zero + one);
        }

        return result;
    }
};

File: Kotlin/1422_Maximum_Score_After_Splitting_a_String.kt
class Solution {
    fun maxScore(s: String): Int {
        var result = 0
        var one = s.count{ it == '1' }
        var zero = 0

        for(i in 0 until s.length - 1) {
            if(s[i] == '0') ++zero
            if(s[i] == '1') --one

            result = maxOf(result, zero + one)
        }

        return result
    }
}